README.md

```diff
@@ -1,12 +1,40 @@
-# my-component
+# Project Migrate
 
-[![Build Status](https://travis-ci.org/keboola/my-component.svg?branch=master)](https://travis-ci.org/keboola/my-component)
+[![Build Status](https://travis-ci.com/keboola/app-project-migrate.svg?branch=master)](https://travis-ci.com/keboola/app-project-migrate)
 
-> Fill in description
+Application which orchestrates whole process of KBC project migration from one KBC stack to Another.
 
-# Usage
+Prerequisites:
+ - Source project which will be migrated
+ - Destination project - empty project where the source project will be cloned
+ 
+Application is executed in *destination* project and requires Storage API token and KBC url of *source* project.
+Admin token of source project is required for GoodData writers migration.
+Source project is left without any changes.
+
+Migration steps performed by the application:
+
+- Create snapshot of source project https://github.com/keboola/app-project-backup
+- Restore project from snapshot https://github.com/keboola/app-project-restore
+- Migrate GoodData writers https://github.com/keboola/app-gooddata-writer-migrate
+- `TODO` Migrate Snowflake writers
+- Migrate Orchestrators https://github.com/keboola/app-orchestrator-migrate
+
+
+## Usage
 
-> fill in usage instructions
+Run the migration in destination project wil the following command.
+This is example of project migration from US to EU, please replace these parameters:
+
+- `DEST_PROJECT_SAPI_TOKEN` - Storage API token associated to admin of destination EU project
+- `SOURCE_PROJECT_SAPI_TOKEN` - Storage API token associated to admin of source US project
+
+```
+curl -X POST \
+ https://docker-runner.eu-central-1.keboola.com/docker/keboola.app-project-migrate/run \
+ -H 'X-StorageApi-Token: DEST_PROJECT_SAPI_TOKEN' \
+ -d '{"configData": {"parameters": {"sourceKbcUrl": "https://connection.keboola.com", "#sourceKbcToken":"SOURCE_PROJECT_SAPI_TOKEN"}}}'
```

Comment:
nemělo by tady bejt ještě zašifrování toho tokenu nebo celýho payloadu, aby nebyl v jobu vidět token?

Reply:
Dělá to samo:
![image](https://user-images.githubusercontent.com/903531/40419232-f5f80f62-5e84-11e8-96d4-7fa75cd9b287.png)

Reply:
jasně! a to jsem zrovna před chvíli koukal na zdroják, co to dělá! omg

Reply:
Ja bych to radeji zasifroval rucne klicem komponenty?
Takhle to muzes vzit a v jakekoliv komponente predat ty data a dostanes ty data rozsifrovany.

Reply:
Treba to poslu nekam genericem v GETu, zapnu si debug mod a dostanu desifrovany credentials/token.

Reply:
klíčem komponenty se to šifruje
https://github.com/keboola/docker-bundle/blob/438421904d1f217c5c9a16e6827e81f0c85fe775/Controller/ApiController.php#L81

Reply:
jenom je tam použitá nejnovější šifra, která v prefixu neudává, jakej klíč se má použít, ale je v payloadu šifry

Reply:
a proč to stejně nezašifruje `https://docker-runner.eu-central-1.keboola.com/docker/encrypt?componentId=keboola.app-project-migrate&projectId=185` který volá ui?
Tam je pak `KBC::ComponentProjectEncrypted==` což je legacy non-KMS context šifra?

Reply:
už to vidím https://github.com/keboola/docker-bundle/blob/438421904d1f217c5c9a16e6827e81f0c85fe775/Resources/config/routing.yml#L2

Reply:
je to součástí migrace šifrování, odin to měl někde sepsaný, nemůžu to teď narychlo najít

Reply:
jinak si myslím že to s tím prefixem není pravda, imho ty prefixy jsou zachovány pořád

Reply:
takže je to v tom prvním případě zašifrováno legacy šifrou https://github.com/keboola/object-encryptor/blob/23374871c1d96b47d282acfea3195cb69c7aa1a5/src/Legacy/Wrapper/BaseWrapper.php#L71
a v tom druhém taky https://github.com/keboola/object-encryptor/blob/23374871c1d96b47d282acfea3195cb69c7aa1a5/src/Legacy/Wrapper/ComponentProjectWrapper.php#L52

Nové šifry jsou např. https://github.com/keboola/object-encryptor/blob/23374871c1d96b47d282acfea3195cb69c7aa1a5/src/Wrapper/ComponentWrapper.php#L51

Reply:
jak se to tam zašifruje se ještě uvidí, to bude součástí dokumentace migrace pro support

Reply:
ale imho je ok to jak to dělá teď ten token/uživatel půjde po migraci pryč

Reply:
j, ten `ComponentSecure` už ale u sebe nic víc nemá, tam už je projekt/komponenta snad fakt uloženo v payloadu :-)

Reply:
v payloadu to nebylo nikdy, u legacy se pomocí toho generuje klíč a u KMS to tvoří encryption context. tzn. rozšifruješ to jenom když nastavíš stejný kontext (komponenta, projekt, konfigurace) jako ten pod kterým to bylo zašifrovaný.

Reply:
tvl, tak přestávám čumět okolo a jdu si hledět svýho, dneska mi to nějak nejde :-(